standalone-metastore/DEV-README

```diff
@@ -0,0 +1,40 @@
+This file contains information for developers and testers.
+
+Testing metastore installation and upgrade against databases beyond Derby:
+There are integration tests for testing installation and upgrade of the
+metastore on MySQL (actually MariaDB is used), Oracle, Postgres, and SQLServer.
+These tests are not run by default because they take several minutes each and
+they require the developer to download the JDBC driver for Oracle.
+They are run in the integration-test phase.
+
+Each ITest runs two tests, one that installs the latest version of the
+database and one that installs the latest version minus one and then upgrades
```

Comment:
I think it would the great we the second test installs an old version (possibly the oldest) of schema and then upgrades it to latest. That would give us better coverage. Any particular reason why we have only one less than latest version?

Reply:
I didn't move all of the old versions of Hive schema (pre 2.3.0) over into the standalone metastore, so we can't test them here.  I didn't move them because it seemed to me easier to leave those where they were and just move the new ones going forward.  Since the standalone metastore doesn't support the old style of calling out to individual upgrade scripts, moving them is not as simple as just copying them.  We actually have to copy and paste the individual 0XX files into each upgrade script.  It's doable if we decide to, but it made more sense to me to say up to 2.3.0 belong to Hive proper, and new stuff belongs to standalone-metastore.

Reply:
Any particular reason why standalone-metastore doesn't support old style of upgrade using patch scripts? I think we should maintain the existing upgrade path if possible. External applications could be linking in to the schemaTool to handle the schema upgrades. Also, I think we should support users to upgrade from Hive 2.0 to move to standalone metastore schema since not all distributions rebase to the latest release of Apache Hive.

Reply:
HiveSchemaTool uses beeline to parse the upgrade files.  beeline lives in and depends on HIveServer2 so the metastore doesn't have access to it.  Thus I switched to use sqlline (which beeline is based on).  But sqlline does not support the nesting of script files.  When I looked at what beeline does to support that it turns out to be a lot of code.  And I couldn't see any reason to keep supporting it.  I don't know why in the past each update was in a separate file, but I see no advantage to it.  So I didn't pull in all the extra code to support it.

I propose a compromise position.  I'll copy all of the upgrade scripts from 1.3.0 through 3.0.0 into standalone metastore.  Then users from the top of 1 and anywhere in 2 can upgrade to standalone metastore.  It shouldn't be too hard for users further back in the 1 line to upgrade to 1.3 first.  Reasonable?